blockchain/indexers/slpindex_test.go

```diff
@@ -0,0 +1,138 @@
+package indexers
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/gcash/bchd/chaincfg/chainhash"
+	"github.com/gcash/bchd/wire"
+	"github.com/simpleledgerinc/goslp"
+	"github.com/simpleledgerinc/goslp/v1parser"
+)
+
+// TestSlpMesesageUnitTests downloads SLP parser unit tests and checks the parser throws for each test where code != nil
+func TestSlpMessageUnitTests(t *testing.T) {
+	resp, err := http.Get("https://raw.githubusercontent.com/simpleledger/slp-unit-test-data/master/script_tests.json")
+	if err != nil {
+		t.Fatal("cannot download unit tests")
+	}
+	data, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	type TestCase struct {
+		Msg    string
+		Script string
+		Code   *float64
+	}
+	var tests []TestCase
+	err = json.Unmarshal(data, &tests)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	for _, test := range tests {
+		slpbuf, _ := hex.DecodeString(test.Script)
+		_, err := v1parser.ParseSLP(slpbuf)
+		if err != nil {
+			if test.Code != nil {
+				continue
+			}
+			t.Fatal("goslp parser did not throw an error")
+		}
+	}
+}
+
+// TestSlpInputUnitTests downloads SLP input unit tests and checks the input conditions for each test are met
+func TestSlpInputUnitTests(t *testing.T) {
+	resp, err := http.Get("https://raw.githubusercontent.com/simpleledger/slp-unit-test-data/master/tx_input_tests.json")
+	if err != nil {
+		t.Fatal("cannot download unit tests")
+	}
+	data, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	type TxItem struct {
+		Txid  string
+		Tx    string
+		Valid bool
+	}
+	type TestCase struct {
+		Description string
+		When        []TxItem
+		Should      []TxItem
+	}
+	var tests []TestCase
+	err = json.Unmarshal(data, &tests)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	for _, test := range tests {
+
+		// create temporary db of input conditions
+		_entryDb := make(map[[32]byte]*SlpIndexEntry)
```

Comment:
In Go we generally don't want variables starting with a _

Reply:
done.